sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubClientBuilder.java

```diff
@@ -688,18 +685,27 @@ private ConnectionOptions getConnectionOptions() {
             ? CbsAuthorizationType.SHARED_ACCESS_SIGNATURE
             : CbsAuthorizationType.JSON_WEB_TOKEN;
 
-        final ClientOptions options = clientOptions != null ? clientOptions : new ClientOptions();
         final SslDomain.VerifyMode verificationMode = verifyMode != null
             ? verifyMode
             : SslDomain.VerifyMode.VERIFY_PEER_NAME;
 
+        final ClientOptions options = new ClientOptions();
+        if (clientOptions != null) {
+            options.setApplicationId(clientOptions.getApplicationId())
+                .setHeaders(clientOptions.getHeaders());
+        }
```

Comment:
Is this needed? I don't think `setHeaders` will make a copy of the headers. So, it's just creating another reference to the same header list.